test/unit/gateways/place_to_pay_test.rb

```diff
@@ -0,0 +1,306 @@
+require 'test_helper'
+
+class PlaceToPayTest < Test::Unit::TestCase
+  def setup
+    @gateway = PlaceToPayGateway.new(login: 'login', secret_key: 'secret_key')
+    
+    @amount = 100
+
+    @payer = {
+      name: "Erika",
+      surname: "Howe",
+      email: "cwilliamson@hotmail.com",
+      documentType: "CC",
+      document: "3572264088",
+      mobile: "3006108300"
+    }
+    payment = {
+      description: 'Cum vitae et consequatur quas adipisci ut rem.',
+      amount: {
+        currency: @gateway.default_currency,
+        total: @amount
+      }
+    }
+    instrument = {
+      card: {
+        installments: 1
+      }
+    }
+
+    @credit_card_approved_visa = credit_card('4110760000000081', month: 12, year: 2023, verification_value: '123', first_name: @payer[:name], last_name: @payer[:surname])
+    @credit_card_rejected_visa = credit_card('4110760000000016', month: 12, year: 2023, verification_value: '123', first_name: @payer[:name], last_name: @payer[:surname])
+
+    @options = {
+      payer: @payer,
+      payment: payment,
+      instrument: instrument
+    }
+
+  end
+
+  def test_successful_purchase
+    @gateway.expects(:ssl_post).returns(successful_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card_approved_visa, @options)
+    assert_success response
+
+    assert_equal '999999', response.authorization
+    assert response.test?
+  end
+
+  def test_failed_purchase
+    @gateway.expects(:ssl_post).returns(failed_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card_rejected_visa, @options)
+    assert_failure response
+    assert_equal Gateway::STANDARD_ERROR_CODE[:card_declined], response.error_code
+  end
+
+  # def test_successful_authorize; end
+
+  # def test_failed_authorize; end
+
+  # def test_successful_capture; end
+
+  # def test_failed_capture; end
+
+  # def test_successful_refund; end
+
```

Comment:
Tests for void and refund need to be written since they are included as transaction types.

Reply:
We added some tests for refund

Reply:
I don't see any unit tests for refund added. Were they left out of a commit? None of the unit tests are currently passing.

Reply:
Tests for refund method were added